### DISCUSSION TITLE
How to enable debug logging?

### DISCUSSION THREAD

**Discussion by hostingnuggets** [netbox-community/netbox-docker#909] [Getting Help]
Hello,

I am trying to debug why the LDAP authentication is not working and do not see any useful log information in the containers themselves. So I was wondering how can I enable debug loggin?

I did not find any environment variable to set or anything in the documentation... Any ideas?

Thanks

**Comment by timrabl** [comment]
Just set `DEBUG=True` in your `env/netbox.env`.

https://github.com/netbox-community/netbox-docker/blob/39c7de4af41703108f8b3af7274bc65813c41799/configuration/configuration.py#L153-L156

**Comment by hostingnuggets** [comment]
Thanks you @timrabl, I did that and restarted all the containers with a docker-compose down and up again but when inspecting the container logs with `docker logs -f netbox-docker_netbox_1` I don't see any more information than usual. Or where exactly is the debug information outputted?

**Reply by timrabl** [reply]
In my case, the netbox container logs (as it should) to `/proc/self/fd/2`, so the equivalent `/dev/stderr` or more known as standard error. This then ends up in the docker container logs.

#### Test 1:

Lets simulate a unreachable LDAP server by changing the LDAP Host in `env/netbox.env`:
```bash
user@srv01 % docker-compose down && docker-compose up -d && docker-compose logs -f netbox netbox-worker | grep -i ldap
netbox_1               | Caught LDAPError while authenticating trst: SERVER_DOWN({'result': -1, 'desc': "Can't contact LDAP server", 'errno': 11, 'ctrls': [], 'info': 'Resource temporarily unavailable'})
```

-> It logs as expected to the container logs.

#### Test 2:

Lets simulate invalid bind user credentials by changing bind user credentials in `env/netbox.env`:
```bash
user@srv01 % docker-compose down && docker-compose up -d && docker-compose logs -f netbox netbox-worker | grep -i ldap
netbox_1               | Caught LDAPError while authenticating trst: INVALID_CREDENTIALS({'msgtype': 97, 'msgid': 1, 'result': 49, 'desc': 'Invalid credentials', 'ctrls': [], 'info': '80090308: LdapErr: DSID-0C090447, comment: AcceptSecurityContext error, data 52e, v3839'})
```

-> It logs as expected to the container logs.

The only difference I can think of could be the `LOGLEVEL` variable, which I set to "debug".  
As far as I know, it is not loaded automatically, this must first be activated here by uncommenting the commented-out line.
https://github.com/netbox-community/netbox-docker/blob/676ca798cd684ce30fca2692180e680d2becf76e/configuration/logging.py#L6

The `logging.py` must of course be adjusted in the container or be in the container via bind mount or custom image.

**Comment by 1-cameron** [comment]
@hostingnuggets  Did you manage to resolve this? I'm having the same issue - container logs not being any more verbose despite enabling debug logging.

**Reply by mongrelion** [reply]
I just managed to get netbox into debugging mode by adding this environment variable to my docker-compose file:
```bash
DEBUG=True
```
Now I even get the stack trace in the browser when something goes wrong.
